pypeit/core/flexure.py

```diff
@@ -432,135 +628,115 @@ def spec_flexure_slit(slits, slitord, slit_bpm, sky_file, method="boxcar", speco
+                # get the dict where the med shift is
+                fdict = flex_list[islit_med_shift].copy()
+
+                # assign fdict to the failed slits
+                for sidx in return_later_slits:
+                    # Reset the dict
+                    flex_dict = copy.deepcopy(empty_flex_dict)
                     # Update dict
-                    for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec', 'sky_spec']:
-                        flex_dict[key].append(fdict[key])
-                    flex_dict['method'].append("boxcar")
-                elif fdict is None and sv_fdict is None:
-                    return_later_sobjs.append(ss)
-                    msgs.warn("Flexure shift calculation failed for this spectrum.")
-                    msgs.info("Will come back to this spectrum to attempt "
-                              "to use saved estimates from other slit/objects")
-
-            # Check if we need to go back
-            if len(return_later_sobjs) > 0:
-                msgs.warn(f'Flexure shift calculation failed for {len(return_later_sobjs)} '
-                          f'object(s) in slit {slits.spat_id[islit]}')
-                for obj_idx in return_later_sobjs:
-                    # Do we have a previously saved value?
-                    if sv_fdict is None:
-                        msgs.warn(f'No previously saved flexure shift estimate available. '
-                                  f'Flexure corrections cannot be performed for object # {obj_idx}')
-                        # Update dict
-                        for key in ['polyfit', 'shift', 'subpix', 'corr', 'corr_cen', 'smooth', 'arx_spec', 'sky_spec']:
-                            # insert None at the location of the object that failed the calculation
-                            flex_dict[key].insert(obj_idx, None)
-                        flex_dict['method'].insert(obj_idx, None)
-                    else:
-                        # Copy me
-                        fdict = copy.deepcopy(sv_fdict)
-                        msgs.info(f"Previously saved flexure shift estimate of {fdict['shift']:.3f} pixels "
-                                  f"used for object # {obj_idx}")
+                    for key in keys_to_update[:-1]:
+                        flex_dict[key].append(fdict[key][0])
+                    # Interpolate
+                    sky_wave_new = flexure_interp(fdict['shift'][0], slit_specs[sidx].wavelength.value)
+                    flex_dict['sky_spec'].append(xspectrum1d.XSpectrum1D.from_tuple((sky_wave_new, slit_specs[sidx].flux.value)))
+
+                    # insert flex_dict in flex_list at the location of the slit that failed the calculation
+                    flex_list[sidx] = flex_dict
+            # local flexure
+            else:
+                # get the dict where the med shift is
+                idx_med_shift = np.where(flex_list[islit_med_shift]['shift'] == med_shift)[0][0]
```

Comment:
Could this fail if there are no elements where `flex_list[islit_med_shift]['shift']==med_shift`? My concern is if there are an even number of slits, for example. Would it be safer to find the slit with the closest value to med_shift?